### PR TITLE
Parallel startup of processes in functional tests

### PR DESCRIPTION
**Pull request by stefano-maggiolo**
In my machine, this reduces the startup time (till first submission)
from 21s to 9s.

---
This change is [Reviewable](https://reviewable.io/reviews/cms-dev/cms/655)

### CODE REVIEW

File: cmstestsuite/programstarter.py [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+        t.start()
+
+    @property
+    def running(self):
+        """Return whether the program is live."""
+        self._check()
+        return self.healthy
+
+    def stop(self):
+        """Quit gracefully. Or not."""
```
Comment (veluca93) [inline cmstestsuite/programstarter.py]: Maybe this docstring could be a bit more explicit (why "or not"?)

File: cmstestsuite/programstarter.py [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+from urlparse import urlsplit
+
+from cmstestsuite import CONFIG, FrameworkException, get_cms_config
+
+
+logger = logging.getLogger(__name__)
+
+
+# Maximum number of attempts to check if a service becomes healthy.
+_MAX_ATTEMPTS = 20
```
Comment (veluca93) [inline cmstestsuite/programstarter.py]: Shouldn't this belong to a class?

Comment (stefano-maggiolo) [inline cmstestsuite/programstarter.py]: Maybe, but I'm using it in two classes and you can argue is a constant of the module.

File: cmstestsuite/programstarter.py (hunk 1 above)
Comment (stefano-maggiolo) [inline cmstestsuite/programstarter.py]: Done.

**Pull request merged**